cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+        if(previousColumnList.size()>0) addedColumnList.removeAll(previousColumnList);
+
+        for (String addedColumn : addedColumnList) {
+            LOGGER.debug("addedColumn = " + addedColumn);
+        }
+        return addedColumnList;
+    }
+
+    private void logCurrentSourcePartitions(List<String> sourcePartitionList, String processId, String instanceExecId, String table) {
+        ProcessLog processLog = new ProcessLog();
+        List<ProcessLogInfo> partitionLogInfoList = new ArrayList<>();
+        for (String sourcePartition : sourcePartitionList) {
+            ProcessLogInfo processLogInfo = new ProcessLogInfo();
+            processLogInfo.setAddTs(new Date());
+            processLogInfo.setProcessId(Integer.parseInt(processId));
+            processLogInfo.setMessage(sourcePartition);
+            processLogInfo.setInstanceRef(Long.parseLong(instanceExecId));
+            processLogInfo.setLogCategory("C2C");
+            processLogInfo.setMessageId(table + " partition");
+            partitionLogInfoList.add(processLogInfo);
+        }
+        processLog.logList(partitionLogInfoList);
+    }
+
+    private void logCurrentSourceColumns(List<String> sourceColumnList, String processId, String instanceExecId, String table) {
+        ProcessLog processLog = new ProcessLog();
+        List<ProcessLogInfo> columnLogInfoList = new ArrayList<>();
+        for (String sourceColumn : sourceColumnList) {
+            ProcessLogInfo processLogInfo = new ProcessLogInfo();
+            processLogInfo.setAddTs(new Date());
+            processLogInfo.setProcessId(Integer.parseInt(processId));
+            processLogInfo.setMessage(sourceColumn);
+            processLogInfo.setInstanceRef(Long.parseLong(instanceExecId));
+            processLogInfo.setLogCategory("C2C");
+            processLogInfo.setMessageId(table + " column");
+            columnLogInfoList.add(processLogInfo);
+        }
+        processLog.logList(columnLogInfoList);
+    }
+
+    private void formStageAndDestTableDDLs(Statement st, List<String> sourceColumnList, String sourceDb, String destDb, String table, String sourceStgtable, String bdreTechPartition, String processId,String instanceExecId) throws Exception {
+        ResultSet rsPartitionList = st.executeQuery("desc "+table);
+        int index = 0;
+        StringBuffer partitionList = new StringBuffer("");
+        List<String> sourcePartitionColumnList = new ArrayList<>();
+        while (rsPartitionList.next()) {
+            if (rsPartitionList.getString(1).equals("# Partition Information")) {
+                index++;
+            }
+            if (index > 0) index++;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00122)